### PR TITLE
Move kubectl apply to staging

### PR DESCRIPTION
**Pull request by Liujingfang1**
**What type of PR is this?**
> /kind cleanup

**What this PR does / why we need it**:
Removing kubectl dependency on kubernetes/kubernetes has been going on since one year ago.
https://github.com/kubernetes/kubectl/issues/80
While the progress is towards the end, we start to move the kubectl packages under staging directory.

This PR moves pkg/kubectl/cmd/apply into staging. Apply is commonly used and users may want to create their own
client side tool or controllers for apply from some libraries. Moving kubectl/apply to staging and then having it
 synced to kubernetes/kubectl repo will make the apply package publicly available.

**Special notes for your reviewer**:
There are several packages that are moved together with apply:
  - pkg/kubectl/cmd/delete
  - pkg/kubectl/cmd/testing
  - pkg/kubectl/cmd/wait
  - pkg/kubectl/cmd/util
  - pkg/kubectl/util
  - pkg/kubectl/generated
  - pkg/kubectl/scheme
  - pkg/kubectl/validation

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/assign @seans3 @pwittrock

### CODE REVIEW

**Comment by BenTheElder** [comment]
To be clear the list of packages in the PR body are all previously under `pkg/kubectl/...` ?

**Comment by seans3** [comment]
Need to run `hack/update-bazel.sh`

**Comment by Liujingfang1** [comment]
> To be clear the list of packages in the PR body are all previously under pkg/kubectl/... ?

Yes.

File: staging/publishing/rules.yaml [hunk 1]
```diff
@@ -877,6 +877,22 @@ rules:
         branch: release-1.14
       - repository: component-base
         branch: release-1.14
+- destination: kubectl
```
Comment (soltysh) [inline staging/publishing/rules.yaml]: We should not update the publishing rules, yet. There's still some code in our kubectl repo which does not exist in the main k/k code. I want to ensure we bring those back but deprecated and not to break people. Also, the publishing should happen not sooner than when we're close to finishing the move.

**Review by soltysh — CHANGES_REQUESTED** [review]
/lgtm cancel
I have one objection wrt publishing all the rest lgtm.

**Pull request closed without merging**